binary.go

```diff
@@ -50,11 +51,34 @@ func (br *BinaryReference) Find(db *Database) *Binary {
 	return db.Content.Meta.Binaries.Find(br.Value.ID)
 }
 
+// BinaryOption is the option function type for use with Binary structs
+type BinaryOption func(binary *Binary)
+
+// WithKDBXv4Binary can be passed to the Binaries.Add function as an option to ensure
+// that the Binary will follow the KDBXv4 format
+func WithKDBXv4Binary(binary *Binary) {
+	binary.Compressed = w.NewBoolWrapper(false)
+	binary.isKDBX4 = true
+}
+
+// WithKDBXv31Binary can be passed to the Binaries.Add function as an option to ensure
+// that the Binary will follow the KDBXv31 format
+func WithKDBXv31Binary(binary *Binary) {
+	binary.Compressed = w.NewBoolWrapper(true)
+	binary.isKDBX4 = false
+}
+
 // Add appends binary data to the slice
-func (bs *Binaries) Add(c []byte) *Binary {
+// Note: this function should not be used directly, use `Database.AddBinary(c []byte) *Binary` instead
+func (bs *Binaries) Add(c []byte, options ...BinaryOption) *Binary {
```

Comment:
I suggest to mark it as deprecated, so we could move the implementation to the func (db *Database) AddBinary in next version and rid of the BinaryOption.

Reply:
Adapted